### DISCUSSION TITLE
Are you interested by a pull request to add LSI reduction?

### DISCUSSION THREAD

**Discussion by rLannes** [colomemaria/epiScanpy#118] [Ideas]
everything in the title.

LSI Latent semantic index is a data reduction method apply primarily in natural language processing but had given great result in ATAC analyses.

**Comment by DaneseAnna** [comment]
Hi, 

Indeed LSI provides interesting results in scATAC. It is practically speaking a Term Frequency Inverse Document Frequency + a TruncatedSVD. We have implemented it as 2 functions epi.pp.tfidf and epi.pp.lsi. 
If you have suggestions to improve the functions, you are more than welcome to do so. However, I would like to point out that in order to run tfidf you need the raw count and thus cannot really correct for library size. As far as I know, other packages took the approach of removing components that correlate with covariates like the number of reads or open peaks per cell. Which isn't optimal. But this is also possible with episcanpy using epi.pp.correlation_component and epi.pp.filtering_components. 

Cheers, 
Anna

PS: sorry I am only now looking at your PR for the gene activity. I was very busy and forgot about it.

**Reply by rLannes** [reply]
Great I did not see that you implemented it already !
